pkg/apiserver/apiserver_test.go

```diff
@@ -1911,6 +1911,7 @@ func TestDeleteWithOptions(t *testing.T) {
 	if simpleStorage.deleted != ID {
 		t.Errorf("Unexpected delete: %s, expected %s", simpleStorage.deleted, ID)
 	}
+	simpleStorage.deleteOptions.GetObjectKind().SetGroupVersionKind(unversioned.GroupVersionKind{})
```

Comment:
These changes are required because I change the `UseOrCreateObject`. The decoder uses the passed in `into` object instead of creating a new one. In turn the versioning decoder [skips](https://github.com/kubernetes/kubernetes/blob/master/pkg/runtime/serializer/versioning/versioning.go#L161) the conversion so the gvk is kept. I think this behavior is an innocuous bug.